src/community/ogcapi/ogcapi-tiles/src/main/java/org/geoserver/api/tiles/TiledCollectionDocument.java

```diff
@@ -184,8 +178,16 @@ CollectionExtents getExtentFromGridsets(TileLayer tileLayer)
         } else {
             // take the first and reproject...
             String srs = srsSet.iterator().next();
-            GridSubset subset = layer.getGridSubsetForSRS(SRS.getEPSG4326());
-            return getExtentsFromGridSubset(subset);
+
+            try {
+                GridSubset subset = layer.getGridSubsetForSRS(SRS.getSRS(srs));
```

Comment:
This is a functional change. I don't understand the code that well, but I'm guessing this is what was intended. The EPSG 4326 version is in the `if() { }` part of the test, and this is meant to handle the "otherwise" case.

Reply:
Yup